[OSDOCS-15132]Improve the ROSA docs for secure auth flows (XCMSTRAT-600) (#95714)
openshift/openshift-docs: modules/rosa-configure.adoc

== login
There are several methods you can use to log into your Red{nbsp}Hat account using the {product-title} (ROSA) CLI (`rosa`). These methods are described in detail below.

[id="rosa-login-sso_{context}"]
=== Authenticating the ROSA CLI with Red Hat single sign-on

You can log in to the ROSA CLI (`rosa`) with Red{nbsp}Hat single sign-on. Red{nbsp}Hat recommends using the `rosa` command line tool with Red{nbsp}Hat single sign-on, instead of using an offline authentication token.

An offline authentication token is long-lived, stored on your operating system, and cannot be revoked. These factors increase overall security risks and the likelihood of unauthorized access to your account.

> Alternatively, authenticating with the Red{nbsp}Hat single sign-on method automatically sends your ROSA CLI instance a refresh token that is valid for 10 hours. Because this authorization code is unique and temporary, it is more secure and reduces the risk of unauthorized access.

Reviewer: ```suggestion
Alternatively, authenticating with the Red{nbsp}Hat single sign-on method automatically sends your `rosa` instance a refresh token that is valid for 10 hours. This authorization code is unique and temporary meaning it is more secure and reduces the risk of unauthorized access.
```

[IMPORTANT]
====

> The method of authenticating using Red Hat single sign-on will not break any existing automations that rely on offline tokens. Red{nbsp}Hat recommends using link:https://console.redhat.com/iam/service-accounts[services accounts] for automation purposes. If you still need to use offline tokens for automation or other purposes, you can download the OpenShift Cluster Manager API token from the link:https://console.redhat.com/openshift/token[OpenShift Cluster Manager API Token] page.

Reviewer: ```suggestion
The method of authenticating using Red Hat single sign-on does not break any existing automations that rely on offline tokens. Red{nbsp}Hat recommends using link:https://console.redhat.com/iam/service-accounts[services accounts] for automation purposes. If you still need to use offline tokens for automation or other purposes, you can download the OpenShift Cluster Manager API token from the link:https://console.redhat.com/openshift/token[OpenShift Cluster Manager API Token] page.
```

====

> * If your system has a web browser, see the "Authenticating the ROSA CLI with a single sign-on authorization code" section to authenticate with Red Hat single sign-on.

Reviewer: ```suggestion
+
Use one of the following methods of authentication:
+
* If your system has a web browser, see the "Authenticating the ROSA CLI with a single sign-on authorization code" section to authenticate with Red{nbsp}Hat single sign-on.
```
Adding a lead-in sentence for the bullet points here, but you may like to modify it.

Author: Good call!

* If you are working with containers, remote hosts, or other environments without a web browser, see the "Authenticating the ROSA CLI with a single sign-on device code" section to authenticate with Red Hat single sign-on.

* To authenticate the ROSA CLI using an offline token, see the "Authenticating the ROSA CLI with an offline token" section.

[NOTE]
====
Single sign-on authorization is supported with ROSA CLI (`rosa`) version 1.2.36 or later.
====

[id="rosa-login-sso_auth{context}"]
=== Authenticating the ROSA CLI with a single sign-on authorization code

> * To log into the ROSA CLI (`rosa`) with a Red{nbsp}Hat single sign-on authorization code, run the following command:

Reviewer: ```suggestion
* To log in to the ROSA CLI (`rosa`) with a Red{nbsp}Hat single sign-on authorization code, run the following command:
```
Agree with the bot - from the IBM style guide: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#l.

Author: Always forget this!

$ rosa login --use-auth-code
----
+

> Running this command will redirect you to the Red{nbsp}Hat single sign-on login. Log in with your Red{nbsp}Hat login or email.

Reviewer: ```suggestion
Running this command redirects you to the Red{nbsp}Hat single sign-on login. Log in with your Red{nbsp}Hat login or email address.
```

To switch accounts, logout from link:https://sso.redhat.com[https://sso.redhat.com] and run the `rosa logout` command in your terminal before attempting to login again.

[id="rosa-login-sso-device_{context}"]
=== Authenticating the ROSA CLI with a single sign-on device code
If you are working with containers, remote hosts, and other environments without a web browser, you can use a Red{nbsp}Hat single sign-on device code for secure authentication. To do this, you must use a second device that has a web browser to approve the login.
[NOTE]
====
Single sign-on authorization is supported with ROSA CLI (`rosa`) version 1.2.36 or later.
====

> * To log into the ROSA CLI (`rosa`) with a Red Hat single sign-on device code, run the following command:

Reviewer: ```suggestion
* To log in to the ROSA CLI (`rosa`) with a Red{nbsp}Hat single sign-on device code, run the following command:
```

[id="rosa-login-token_{context}"]
=== Authenticating the ROSA CLI with an offline token

> Log into your Red{nbsp}Hat account, saving the credentials to the `rosa` configuration file.

Reviewer: ```suggestion
Log in to your Red{nbsp}Hat account, saving the credentials to the `rosa` configuration file.
```

[NOTE]
====
To use offline tokens for automation purposes, you can download the OpenShift Cluster Manager API token from the link:https://console.redhat.com/openshift/token/rosa[OpenShift Cluster Manager API Token] page.
To use service accounts for automation purposes, see the link:https://console.redhat.com/iam/service-accounts[Service Accounts] page.
====

[IMPORTANT]
====
Red{nbsp}Hat recommends using service accounts for automation purposes.
====

> * To log into ROSA CLI (`rosa`) with a Red{nbsp}Hat offline token, run the following command:

Reviewer: ```suggestion
* To log in to ROSA CLI (`rosa`) with a Red{nbsp}Hat offline token, run the following command:
```
